### ISSUE TITLE
Enhancement: [return-await] Option for unopinionated "in-try-catch"

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#9030]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/return-await

### Description

The current `in-try-catch` option handles two separate concerns 
- _Correctness_ - requiring `await` where it impacts control flow (due to `try-catch-finally`)
- _Stylistic_ - _forbidding_ `await` elsewhere

I would propose for there to be a rule option that handles only the correctness portion. That is to say, it should enforce `await` wherever in-try-catch currently does, but not report where `in-try-catch` forbids `await`.

### Fail

```TypeScript
// same as in-try-catch
async function f() {
  try {
    return returnsPromise();
  } catch { }
}
```

### Pass

```TypeScript
async function f() {
   return returnsPromise();
}

async function f() {
   return await returnsPromise();
}

// same as in-try-catch
async function f() {
  try {
    return await returnsPromise();
  } catch { }
}
```

### Additional Info

I think this would be valuable in order to be able to recommend a less disruptive version of the rule in a preset, see https://github.com/typescript-eslint/typescript-eslint/issues/8667

**Comment by kirkwaiblinger** [comment]
I'm not quite sure what to name this option, since, unfortunately, `in-try-catch` is already in use to mean something different. Maybe `in-try-catch-only`? 🤷

**Comment by bradzacher** [comment]
I'm a -1 here.

What does one gain by allowing `return×await`? What does the laxness enable?

All I can personally see is a big downside - allowing the inconsistency creates a confusing code standard - "Why is this function `return await`, but this one isn't? What's the correct code to write?"

I don't think there are any users that are evaluating this rule and this option, and skipping it because it's technically got some stylistic implications. In fact I'd say the opposite - people like that it enforces strict consistency.

**Comment by kirkwaiblinger** [comment]
I don't feel strongly about it... I would personally only ever use "always". 

> What does one gain by allowing `return await`? What does the laxness enable?

This minimizes the "big diff bad" complaint when newly enabling something in a codebase, and differentiates between signal and noise (control flow changes vs consistency changes) when reviewing the changeset. Mostly, my thought is that this is a means to an end to make it palatable for the `recommended-type-checked` config, where we may be recommending it to existing mixed/always/never-except-for-error-handling codebases that were not previously linted.
  
> I don't think there are any users that are evaluating this rule and this option, and skipping it because it's technically got some stylistic implications. In fact I'd say the opposite - people like that it enforces strict consistency.

Thus far, I just have existence proof of one person in the wild (and who isn't me 😄) who asked for this as a prerequisite for adoption.
 
> Consistency is good for style _and correctness_ because it means people think less - allowing them to get it right more often. "never use `return await` unless it's in a `try`" is simple and easy to teach. But if people start seeing code that often violates that then it muddies the water.

Agreed! Again, for which reason I'd use `always`. But, tbh I'd still use the proposed option as a stepping stone to make the diff safer. 

Another approach here might be restructuring the options as 
"affectsControlFlow": "always" | "never" | "dontCare",
"doesNotAffectControlFlow": "always" | "never" | "dontCare"
but that seems needlessly breaking 🤷

**Comment by kirkwaiblinger** [comment]
Not much enthusiasm for this, and it was decided it's not going to be used as an enabler for #8667. Closed!

**Comment by kirkwaiblinger** [comment]
reopened a la https://github.com/typescript-eslint/typescript-eslint/issues/8667#issuecomment-2148569319
